stream/BidirectionalStream.java

```diff
@@ -68,14 +68,14 @@ public BidirectionalStream(TypeDBStub stub, RequestTransmitter transmitter) {
         ResponseCollector.Queue<Res> queue = resCollector.queue(requestID);
         if (batch) dispatcher.dispatch(req);
         else dispatcher.dispatchNow(req);
-        return new Single<>(queue);
+        return new Single<>(requestID, queue, this);
     }
 
     public Stream<ResPart> stream(Req.Builder request) {
         UUID requestID = UUID.randomUUID();
         ResponseCollector.Queue<ResPart> collector = resPartCollector.queue(requestID);
         dispatcher.dispatch(request.setReqId(UUIDAsByteString(requestID)).build());
-        ResponsePartIterator iterator = new ResponsePartIterator(requestID, collector, dispatcher);
+        ResponsePartIterator iterator = new ResponsePartIterator(requestID, collector, this);
```

Comment:
now add the stream to the iterator/Single so they can call back when they are finished in order to clean up